test_conformance/api/test_queries.cpp

```diff
@@ -273,50 +364,78 @@ int test_get_command_queue_info(cl_device_id deviceID, cl_context context, cl_co
     clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_ON_HOST_PROPERTIES, sizeof(device_props), &device_props, NULL);
     log_info("CL_DEVICE_QUEUE_ON_HOST_PROPERTIES is %d\n", (int)device_props);
 
-    // Mask off vendor extension properties.  Only test standard OpenCL properties
-    device_props &= CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE|CL_QUEUE_PROFILING_ENABLE;
+    auto version = get_device_cl_version(deviceID);
 
-    queue_props[1] = device_props;
-    clCommandQueueWrapper queue = clCreateCommandQueueWithProperties( context, deviceID, &queue_props[0], &error );
-    test_error( error, "Unable to create command queue to test with" );
-
-    cl_uint refCount;
-    error = clGetCommandQueueInfo( queue, CL_QUEUE_REFERENCE_COUNT, sizeof( refCount ), &refCount, &size );
-    test_error( error, "Unable to get command queue reference count" );
-    if( size != sizeof( refCount ) )
+    int num_test_options = MIN_NUM_COMMAND_QUEUE_PROPERTIES;
+    if (device_props & CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE)
     {
-        log_error( "ERROR: Returned size of command queue reference count does not validate! (expected %d, got %d)\n", (int)sizeof( refCount ), (int)size );
-        return -1;
+        // Test out-of-order queues properties if supported
+        num_test_options = OOO_NUM_COMMAND_QUEUE_PROPERTIES;
+        if (version > Version(2, 0) && !is_compatibility)
+        {
+            // Test queue on device if supported
+            num_test_options = ARRAY_SIZE(property_options);
```

Comment:
This doesn't look quite right - OpenCL 3.0 devices are not required to support on-device queues IIRC.